posts/2023-10-20-open-liberty-with-cargo-tracker-ee10.adoc

```diff
@@ -0,0 +1,126 @@
+
+* <<maven_setup, Maven command line setup>> 
+* <<vscode_setup, Visual Studio Code IDE setup>>
+
+https://openliberty.io/docs/latest/development-mode.html[Liberty dev mode] is used in both setup. Because it provides the following three key features, you can deeply learn Jarakee EE through debugging or updating the Cargo Tracker code:
+
+* Automatically detect, recompile, and deploy code changes in your running Liberty.
+* Unit and integration tests are run on demand when you press `Enter` in the command terminal.
+* Finally, you can attach a debugger to the running Liberty to step through the code.
+
+
+
+[#maven_setup]
+== Maven command line setup
+
+After cloning the project, you can quickly start Cargo Tracker through the command line with Maven. Navigate to the project base directory (where the pom.xml file is located) and run:
+
+[source]
+----
+mvn -Popenliberty liberty:dev
+----
+ 
+The application should start without any thrown exceptions. You can safely ignore the shrinkwrap features warning, message-driven bean warning, and the AggregateObjectMapping nested foreign key warning, as they don't affect the application functionality.
+
+image::/img/blog/cargo-tracker-ee10-dev-mode.png[Cargo Tracker Maven Output,width=70%,align="center"]
+
+
+[#vscode_setup]
+== Visual Studio Code IDE setup
+
+We'd also recommend installing the Liberty Tools for Visual Studio Code extension in your IDE to benefit from the fast and iterative development features it offers directly within the IDE. You can download this extension directly from the https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[VS Code marketplace].
+
+After you start VSCode IDE, open the Cargo Tracker project by selecting **Explorer** > **Open Folder**. Select the `cargotracker` directory.
+
+image::/img/blog/cargo-tracker-vscode-open-folder.png[Open Cargo Tracker project,width=70%,align="center"]
+
+The `cargo-tracker` project appears under the `LIBERTY DASHBOARD` extension in the Explorer sidebar. To run Cargo Tracker, right-click the `cargo-tracker` project and select the **Start...** option.
+
+image::/img/blog/cargo-tracker-vscode-start.png[Start Cargo Tracker,width=70%,align="center"]
+
+You can then run dev mode with the Open Liberty profile active by typing in `-Popenliberty` in the pop-up entry field and then press the `enter/return` key. 
+
+image::/img/blog/cargo-tracker-vscode-parameters.png[Start Cargo Tracker with dash P parameter,width=60%,align="center"]
+
+After starting with the custom parameter for the Open Liberty profile, a terminal window in the IDE opens and the https://github.com/OpenLiberty/ci.maven#readme[Liberty Maven plug-in] starts the application. Liberty is running in dev mode. You can safely ignore the shrinkwrap features warning, message-driven bean warning, and the AggregateObjectMapping nested foreign key warning, as they don't affect the application functionality.
```

Comment:
Liberty is now running in dev mode - maybe add now to show how it links to previous sentence?